lib/enigma.rb

```diff
@@ -37,4 +37,35 @@ def shifts(key, date)
     end
     shifts
   end
+
+  def to_indexes(string)
+    alphabet = Array("a".."z")<< " "
+    string_array = string.downcase.chars
+    string_array.map do |char|
+      alphabet.find_index(char)
+    end
+  end
+
+  def indexes_to_string(string, shifts)
+    alphabet = Array("a".."z")<< " "
+    shifts.map do |number|
+      alphabet.rotate(number)[0]
+    end.join
+  end
+
+  def cipher(string, shifts)
+    indexes_plus_shifts = []
+    to_indexes(string).each_with_index do |number, index|
+      indexes_plus_shifts << number + shifts[index%4]
+    end
+    indexes_to_string(string, indexes_plus_shifts)
+  end
+
+  def decipher(string, shifts)
+    indexes_plus_shifts = []
+    to_indexes(string).each_with_index do |number, index|
+      indexes_plus_shifts << number - shifts[index%4]
+    end
+    indexes_to_string(string, indexes_plus_shifts)
+  end
```

Comment:
`decipher` works much the same as `cipher` the notable difference is that instead of adding together each number in shifts to each number in our indexes, we instead subtract our shifts from our indexes. 
This will result in our original indexes array being returned.
Once we call `indexes_to_string` on this original indexes array the return value will be our deciphered string, down cased of course.